src/controllers/role.controller.js

```diff
@@ -0,0 +1,41 @@
+import responseHandlers from '../utils/responseHandlers';
+import statusCodes from '../utils/statusCodes';
+import customMessages from '../utils/customMessages';
+import roles from '../services/role.service';
+import authentication from '../services/authentication.service';
+import utils from '../utils/authentication.utils';
+
+const { decodeToken } = utils;
+const { assignRole } = roles;
+const { findUserByEmailOrUsername } = authentication;
+const { userNotAllowedForAction, notExistUser, roleAssigned, existingRole } = customMessages;
+const { unAuthorized, forbidden, ok } = statusCodes;
+
+/**
+   * @description user role controller class
+   */
+export default class UserRoleController {
+      /**
+   * @param {Request} req Node/express request
+   * @param {Response} res Node/express response
+   * @returns {Object} Custom response depending on function's inputs
+   * @description update user role
+   */
+    static assignRole = async (req, res) => {
+        const token = req.headers.authorization.split(' ')[1];
+        const tokenDecoded = await decodeToken(token);
```

Comment:
I don't think it's a good idea to `decode the token` inside the controller because there is a middleware function called `isLoggedInAndVerified` that decodes the token a return the user in this variable `req.sessionUser` so here you can do this `const tokenDecoded = req.sessionUser` because I saw that you call that middleware `isLoggedInAndVerified` in your routes. maybe you have a reason for doing this instead of using the middleware, I would like to hear it.

Reply:
Yes for sure, I saw that too, and thanks for remembering this, she should have used session user which is saved in the request object instead of hard coding it again, it is a kind of redudancy